pkg/skaffold/deploy/kustomize_test.go

```diff
@@ -17,11 +17,118 @@ limitations under the License.
 package deploy
 
 import (
+	"context"
+	"io/ioutil"
 	"testing"
 
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/GoogleContainerTools/skaffold/testutil"
+	"github.com/pkg/errors"
 )
 
+func TestKustomizeDeploy(t *testing.T) {
+	tmpDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	var tests = []struct {
+		description string
+		cfg         *latest.KustomizeDeploy
+		builds      []build.Artifact
+		command     util.Command
+		shouldErr   bool
+	}{
+		{
+			description: "no manifest",
+			cfg: &latest.KustomizeDeploy{
+				KustomizePath: tmpDir.Root(),
+			},
+			command: testutil.NewFakeCmd(t).
+				WithRunOut("kubectl version --client -ojson", kubectlVersion).
+				WithRunOut("kustomize build "+tmpDir.Root(), ""),
+		},
+		{
+			description: "deploy success",
+			cfg: &latest.KustomizeDeploy{
+				KustomizePath: tmpDir.Root(),
+			},
+			command: testutil.NewFakeCmd(t).
+				WithRunOut("kubectl version --client -ojson", kubectlVersion).
+				WithRunOut("kustomize build "+tmpDir.Root(), deploymentWebYAML).
+				WithRun("kubectl --context kubecontext --namespace testNamespace apply --force -f -"),
```

Comment:
not a huge deal, but can you use the variables `testKubeContext` and `testNamespace` here instead of the strings?

```golang
WithRun("kubectl --context " + testKubeContext + " -- namespace " + testNamespace + " apply --force -f -")
```
or 
```golang
WithRun(fmt.Sprintf("kubectl --context %s --namespace %s apply --force -f -", testKubeContext, testNamespace))
```

Reply:
I don't know about that. Those strings are basically assertions and I'd rather hard code the expected result than compute it. I know it's debatable...

Reply:
ah ok I misunderstood, I thought you were actually generating the command to run here. it does feel weird to me that this test will break if we rename a variable somewhere...but I see where you're coming from. I don't feel strongly enough about it to push for you to change it :)

Reply:
Thanks!